lib/dangermattic/plugins/manifest_pr_checker.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Danger
+  # Plugin to check if the Gemfile.lock was updated when changing the Gemfile in a PR.
+  class ManifestPRChecker < Plugin
+
+    # Check if the `Gemfile` file was modified without a corresponding `Gemfile.lock` update 
+    def check_gemfile_lock_updated
+      check_manifest_lock_updated(
+        file_name: 'Gemfile',
+        lock_file_name: 'Gemfile.lock',
+        instruction: 'Please run `bundle install` or `bundle update <updated_gem>`'
+      )
+    end
+
+    # Check if the `Package.swift` file was modified without a corresponding `Package.resolved` update 
+    def check_swift_package_resolved_updated
+      check_manifest_lock_updated(
+        file_name: 'Package.swift',
+        lock_file_name: 'Package.resolved',
+        instruction: 'Please resolve the Swift packages in Xcode'
+      )
+    end
+
+    private
+
+    def check_manifest_lock_updated(file_name:, lock_file_name:, instruction:)
+      manifest_modified = git.modified_files.include?(file_name)
+      lock_modified = git.modified_files.include?(lock_file_name)
+      
+      if manifest_modified && !lock_modified
+        failure("#{file_name} was changed without updating #{lock_file_name}. #{instruction}.")
+      end
```

Comment:
I'm wondering if there could be cases where the manifest file was modified but the lockfile wouldn't.

For example, if you update your `Package.swift` to change tabs to spaces or fix some indentation or add blank lines for formatting or add/remove comments… I think the generated `Package.resolved` would still end up unmodified?

And it might be the same for `Gemfile.lock`?

(It's not the case for `Podfile.lock` because those contain a `PODFILE CHECKSUM` field at the end of the lockfile, so even if the `Podfile` file changed without changing semantics or the list of pods nor their version reauirements, the `Podfile.lock` would still be modified, at the very least to have that checksum updated)

Reply:
> I'm wondering if there could be cases where the manifest file was modified but the lockfile wouldn't.

Good point -- I thought about that as well, but while I believe this indeed will be the case, it tends to be somewhat rare. Still annoying to have a false negative validation. A couple of alternatives:
1. Have just a warning instead, which can just be ignored when that happens. It is the simples solution, but in a situation where the files are actually out of sync and the warning is there, it might become _too easy_ to ignore it.
2. Make the change comparison a bit more sophisticated, which is more work to implement (I haven't fully thought it out to consider complexity and all cases, but it is certainly a lot more complex than just checking "both files were changed") but will make the plugin more reliable and correct.

I'm tempted to just go with 1 for now, but I can also try quickly looking into it to have some idea of how complex it would be -- perhaps it wouldn't be so complicated after all. For the `Gemfile`, we could probably use something from Bundler such as `Bundler::Definition` and `Bundler::LockfileParser`, but for `Package.swift` I guess it would be a bit more complicated 🤔

Reply:
Update: I've just changed the code to emit warnings for now. I think we could also consider this version "feature complete", and perhaps it is worth it to brainstorm some ideas for making the check smarter (and more precise) on a second iteration.